packages/filesystem/src/browser/file-dialog/file-dialog.ts

```diff
@@ -25,18 +25,40 @@ import { FileDialogModel } from './file-dialog-model';
 import { FileDialogWidget } from './file-dialog-widget';
 import { FileDialogTreeFiltersRenderer, FileDialogTreeFilters } from './file-dialog-tree-filters-renderer';
 
-export const FileDialogFactory = Symbol('FileDialogFactory');
-export interface FileDialogFactory {
-    (props: FileDialogProps): FileDialog;
+export const OpenFileDialogFactory = Symbol('OpenFileDialogFactory');
+export interface OpenFileDialogFactory {
+    (props: OpenFileDialogProps): OpenFileDialog;
+}
+
+export const SaveFileDialogFactory = Symbol('SaveFileDialogFactory');
+export interface SaveFileDialogFactory {
+    (props: SaveFileDialogProps): SaveFileDialog;
 }
 
 export const NAVIGATION_PANEL_CLASS = 'theia-NavigationPanel';
 export const CONTROL_PANEL_CLASS = 'theia-ControlPanel';
 
-@injectable()
 export class FileDialogProps extends DialogProps {
+
+    /**
+     * A set of file filters that are used by the dialog. Each entry is a human readable label,
+     * like "TypeScript", and an array of extensions, e.g.
+     * ```ts
+     * {
+     * 	'Images': ['png', 'jpg']
+     * 	'TypeScript': ['ts', 'tsx']
+     * }
+     * ```
+     */
+    filters?: FileDialogTreeFilters;
```

Comment:
With the current code, "All Files" always stays at the top when the dialog opens, playing the role as the default filter. IMO, it would be nice if the API offers users the flexibility of deciding which filter the file dialog to start with :)

What do you think?

Reply:
I haven't thought about this before. Just added 'All Files' at the top of the list and got working code. But it's a great thing to improve that.